Sources/MapboxMaps/Foundation/MapProjectionDelegate.swift

```diff
@@ -0,0 +1,75 @@
+import CoreLocation
+
+internal protocol MapProjectionDelegate: AnyObject {
+    /// Calculate distance spanned by one pixel at the specified latitude and
+    /// zoom level.
+    ///
+    /// - Parameters:
+    ///   - latitude: The latitude for which to return the value
+    ///   - zoom: The zoom level
+    ///
+    /// - Returns: Meters
+    static func metersPerPoint(for latitude: CLLocationDegrees, zoom: CGFloat) -> Double
+
+    /// Calculate Spherical Mercator ProjectedMeters coordinates.
+    /// - Parameter coordinate: Coordinate at which to calculate the projected
+    ///     meters
+    ///
+    /// - Returns: Spherical Mercator ProjectedMeters coordinates
+    static func projectedMeters(for coordinate: CLLocationCoordinate2D) -> ProjectedMeters
+
+    /// Calculate a coordinate for a Spherical Mercator projected
+    /// meters.
+    ///
+    /// - Parameter projectedMeters: Spherical Mercator ProjectedMeters coordinates
+    ///
+    /// - Returns: A coordinate
+    static func coordinate(for projectedMeters: ProjectedMeters) -> CLLocationCoordinate2D
+
+    /// Calculate a point on the map in Mercator Projection for a given
+    /// coordinate at the specified zoom scale.
+    ///
+    /// - Parameters:
+    ///   - coordinate: The coordinate for which to return the value.
+    ///   - zoomScale: The current zoom factor applied on the map, is used to
+    ///         calculate the world size as tileSize * zoomScale (i.e.
+    ///         512 * 2 ^ Zoom level) where tileSize is the width of a tile
+    ///         in points.
+    /// - Returns: Mercator coordinate
+    static func project(_ coordinate: CLLocationCoordinate2D, zoomScale: CGFloat) -> MercatorCoordinate
+
+    /// Calculate a coordinate for a given point on the map in Mercator Projection.
+    ///
+    /// - Parameters:
+    ///   - mercatorCoordinate: Point on the map in Mercator projection.
+    ///   - zoomScale: The current zoom factor applied on the map, is used to
+    ///         calculate the world size as tileSize * zoomScale (i.e.
+    ///         512 * 2 ^ Zoom level) where tileSize is the width of a tile in
+    ///         points.
+    /// - Returns: Unprojected coordinate
+    static func unproject(_ mercatorCoordinate: MercatorCoordinate, zoomScale: CGFloat) -> CLLocationCoordinate2D
+}
+
+public class Projection: MapProjectionDelegate {
```

Comment:
On Android the `MapboxMap : MapProjectionDelegate` to be more discoverable, where the functions inside MapProjectionDelegate is not static.

gl-native exposed generated static APIs can be directly used as `Projection.project(point, zoomScale)`.

Reply:
Let's leave this as a point of difference? At the moment there's no need on iOS because of the lack of plugins. /cc @tobrun